From 3b5726bf5de370ce942b349c688f7c055c44e906 Mon Sep 17 00:00:00 2001
From: Pedro Hernandez <pehg_@outlook.com>
Date: Thu, 13 Mar 2025 03:03:31 -0400
Subject: [PATCH] Implement horizontal scrolling with Shift + Vertical Scroll
 in `DocumentView`

This change introduces horizontal scrolling functionality when users hold
Shift and scroll vertically using the mouse wheel in the `DocumentView`
widget.
This feature is particularly useful for navigating zoomed-in images,
especially for users relying solely on a mouse with a vertical scroll wheel.
By aligning with the behavior of popular applications like Gimp, Firefox,
and Chrome, this update ensures a more intuitive and consistent user
experience in Gwenview.

FEATURE: 491463
---
 lib/documentview/documentview.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/documentview/documentview.cpp b/lib/documentview/documentview.cpp
index a9bb4910c..26968d64b 100644
--- a/lib/documentview/documentview.cpp
+++ b/lib/documentview/documentview.cpp
@@ -926,7 +926,7 @@ void DocumentView::wheelEvent(QGraphicsSceneWheelEvent *event)
     // 16 = pixels for one line
     // 120: see QWheelEvent::angleDelta().y() doc
     qreal dy = -qApp->wheelScrollLines() * 16 * event->delta() / 120;
-    if (event->orientation() == Qt::Horizontal) {
+    if ((event->orientation() == Qt::Horizontal) || (event->modifiers() & Qt::ShiftModifier)) {
         std::swap(dx, dy);
     }
     d->mAdapter->setScrollPos(d->mAdapter->scrollPos() + QPointF(dx, dy));
-- 
GitLab

